auth/src/desktop/rpcs/auth_request.cc

```diff
@@ -33,54 +35,22 @@ const char* kHeaderFirebaseLocale = "X-Firebase-Locale";
 
 AuthRequest::AuthRequest(::firebase::App& app, const char* schema,
                          bool deliver_heartbeat)
-    : RequestJson(schema) {
-  // The user agent strings are cached in static variables here to avoid
-  // dependencies upon other parts of this library.  This complication is due to
-  // the way the tests are currently configured where each library has minimal
-  // dependencies.
-
-  CheckEmulator();
-  static std::string auth_user_agent;           // NOLINT
-  static std::string extended_auth_user_agent;  // NOLINT
-  static Mutex* user_agent_mutex = new Mutex();
-  MutexLock lock(*user_agent_mutex);
-  if (auth_user_agent.empty()) {
-    std::string sdk;
-    std::string version;
-    app_common::GetOuterMostSdkAndVersion(&sdk, &version);
-    // Set the user agent similar to the iOS SDK.  Format:
-    // FirebaseAuth.<platform>/<sdk_version>
-    assert(!(sdk.empty() || version.empty()));
-    std::string sdk_type(sdk.substr(sizeof(FIREBASE_USER_AGENT_PREFIX) - 1));
-    auth_user_agent = std::string("FirebaseAuth.") + sdk_type + "/" + version;
-    // Generage the extended header to set the format specified by b/28531026
-    // and b/64693042 to include the platform and framework.
-    // <environment>/<sdk_implementation>/<sdk_version>/<framework>
-    // where <framework> is '(FirebaseCore|FirebaseUI)'.
-    extended_auth_user_agent = std::string(app_common::kOperatingSystem) + "/" +
-                               sdk + "/" + version + "/" + "FirebaseCore-" +
-                               sdk_type;
-  }
-  // TODO(b/244643516): Remove the User-Agent and X-Client-Version headers.
-  if (!auth_user_agent.empty()) {
-    add_header("User-Agent", auth_user_agent.c_str());
-    add_header("X-Client-Version", extended_auth_user_agent.c_str());
-  }
-  if (deliver_heartbeat) {
-    std::shared_ptr<heartbeat::HeartbeatController> heartbeat_controller =
-        app.GetHeartbeatController();
-    if (heartbeat_controller) {
-      std::string payload = heartbeat_controller->GetAndResetStoredHeartbeats();
-      std::string gmp_app_id = app.options().app_id();
-      if (!payload.empty()) {
-        add_header(app_common::kApiClientHeader, payload.c_str());
-        add_header(app_common::kXFirebaseGmpIdHeader, gmp_app_id.c_str());
-      }
-    }
-  }
```

Comment:
This part should keep. This is for the new desktop heartbeat Matt implemented.

Reply:
done